Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs

```diff
@@ -67,71 +69,35 @@ public class DafnyProgramVerifier : IProgramVerifier {
         : Convert.ToInt32(options.VcsCores);
     }
 
-    public async Task<VerificationResult> VerifyAsync(Dafny.Program program,
+    // 256MB
+    private const int MaxStackSize = 0x10000000;
+    static readonly ThreadTaskScheduler LargeStackScheduler = new(MaxStackSize);
+
+    public IReadOnlyList<IImplementationTask> Verify(Dafny.Program program,
                                      IVerificationProgressReporter progressReporter,
                                      CancellationToken cancellationToken) {
-      await mutex.WaitAsync(cancellationToken);
-      try {
-        // The printer is responsible for two things: It logs boogie errors and captures the counter example model.
-        var errorReporter = (DiagnosticErrorReporter)program.reporter;
-        var printer = new ModelCapturingOutputPrinter(logger, errorReporter, progressReporter);
-        // Do not set these settings within the object's construction. It will break some tests within
-        // VerificationNotificationTest and DiagnosticsTest that rely on updating these settings.
-        DafnyOptions.O.TimeLimit = options.TimeLimit;
-        DafnyOptions.O.VcsCores = GetConfiguredCoreCount(options);
-        DafnyOptions.O.Printer = printer;
-
-        var executionEngine = new ExecutionEngine(DafnyOptions.O, cache);
-        var translated = Translator.Translate(program, errorReporter, new Translator.TranslatorFlags {
-          InsertChecksums = true,
-          ReportRanges = true
-        });
-        var moduleTasks = translated.Select(t => {
-          var (moduleName, boogieProgram) = t;
-          var programId = program.FullName;
-          var boogieProgramId = (programId ?? "main_program_id") + "_" + moduleName;
-          return VerifyWithBoogieAsync(TextWriter.Null, executionEngine, boogieProgram, cancellationToken, boogieProgramId);
-        }).ToList();
-        await Task.WhenAll(moduleTasks);
-        var verified = moduleTasks.All(t => t.Result);
-        return new VerificationResult(verified, printer.SerializedCounterExamples);
-      }
-      finally {
-        mutex.Release();
-      }
-    }
 
-    private async Task<bool> VerifyWithBoogieAsync(TextWriter output,
-      ExecutionEngine engine, Boogie.Program program,
-      CancellationToken cancellationToken, string programId) {
-      program.Resolve(engine.Options);
-      program.Typecheck(engine.Options);
-
-      engine.EliminateDeadVariables(program);
```

Comment:
We have four instances of this call chain in Dafny; would it be worth adjusting the other callers as well?

Reply:
I think that requires more updates to the Boogie API.

Reply:
I suspect the differences we have are mostly oversight rather than conscious decisions to run things slightly differently in each case, but in any case we can handle this later.